ZOOKEEPER-3811 : cleaning up code,static field be directly referred by its class name

XavierChengZW opened the pull request:
optimized code,A static field should be directly referred by its class name

eolivelli commented:
Looks good.
Can you please open a JIRA and cite the jira ticket id on the commit message? Then we are good to commit this patch.

Personally I don't think this is really an 'optimized' version of the code but now the code looks cleaner. We should check the actual byte code emitted by javac

It is strange that spotbugs did not quarrel about these accesses to static variables.
Did we put some FBSuppressWarning directive?

XavierChengZW commented:
> Looks good.
> Can you please open a JIRA and cite the jira ticket id on the commit message? Then we are good to commit this patch.
> 
> Personally I don't think this is really an 'optimized' version of the code but now the code looks cleaner. We should check the actual byte code emitted by javac
> 
> It is strange that spotbugs did not quarrel about these accesses to static variables.
> Did we put some FBSuppressWarning directive?

Yeah 

> Looks good.
> Can you please open a JIRA and cite the jira ticket id on the commit message? Then we are good to commit this patch.
> 
> Personally I don't think this is really an 'optimized' version of the code but now the code looks cleaner. We should check the actual byte code emitted by javac
> 
> It is strange that spotbugs did not quarrel about these accesses to static variables.
> Did we put some FBSuppressWarning directive?

OK, I've added a JIRA and changed the title description.  When you are in spotbugs, you can use a variety of code specification plug-ins to scan, which will have a good effect

symat commented:
I would also though that this should have been caught by spotbugs. At least IntellJ usually remind me to these things :)

I'll merge it to master.

symat commented:
merged, thanks for caring about ZooKeeper code! :)
